core/new-gui/src/app/workspace/component/code-editor-dialog/code-editor-dialog.component.ts

```diff
@@ -56,28 +61,68 @@ export class CodeEditorDialogComponent implements AfterViewInit, SafeStyle, OnDe
 
   ngOnDestroy(): void {
     this.workflowActionService.getTexeraGraph().updateSharedModelAwareness("editingCode", false);
+
+    if (this.languageServerSocket !== undefined) {
+      this.languageServerSocket.close();
+      this.languageServerSocket = undefined;
+    }
   }
 
-  ngAfterViewInit() {
+  createLanguageClient(transports: MessageTransports): MonacoLanguageClient {
+    return new MonacoLanguageClient({
+      name: "Python UDF Language Client",
+      clientOptions: {
+        documentSelector: ["python"],
+        errorHandler: {
+          error: () => ({ action: ErrorAction.Continue }),
+          closed: () => ({ action: CloseAction.DoNotRestart }),
+        },
+      },
+      connectionProvider: {
+        get: () => {
+          return Promise.resolve(transports);
+        },
+      },
+    });
+  }
+
+  getOrCreateModel() {
+    const uri = this.getModelURI();
+    return (
+      monaco.editor.getModel(monaco.Uri.parse(uri)) ||
+      monaco.editor.createModel(this.code, "python", monaco.Uri.parse(uri))
+    );
+  }
+
+  getLanguageServerSocket() {
+    return this.languageServerSocket;
+  }
+
+  getModelURI() {
     const currentOperatorId: string = this.workflowActionService
       .getJointGraphWrapper()
       .getCurrentHighlightedOperatorIDs()[0];
+    return `inmemory://${currentOperatorId}.py`;
+  }
 
-    if (currentOperatorId === undefined) {
-      return;
-    }
+  ngAfterViewInit() {
+    const currentOperatorId: string = this.workflowActionService
```

Comment:
Let's pass this currentOperatorId as an `@Input` from the parent component to avoid the case where the user may highlight multiple operators.

Reply:
This will be covered in a latter PR